### PR TITLE
Add feature for adding existing goals to a path - fixes #344

### LINKED ISSUE #344: Adding pre-existing goals to a path
I should be able to add goals that are already existing in the Goals tab to one of my paths.

### PR DESCRIPTION
**Pull request by karolsojko**
Fixes #344 

lut-15-2017 11-01-14

### CODE REVIEW

File: app/actions/PathsActions.js [hunk 1]
```diff
@@ -107,6 +112,40 @@ export function pathsUpdateGoal(path, goal, data, slackOptions = {}) {
...
 
+export function addExistingGoalToPathRequest() {
+  return (dispatch, getState) => {
+    const {
+      name,
+      description,
+      tags,
+      level,
+      icon
+    } = getState().goals.addExistingGoalsModal.selectedGoal;
```
Comment (szymonmichalak) [inline app/actions/PathsActions.js]: Accessing state in action is an anti-pattern and should be avoided if possible. Can you pass those props from component itself?

File: app/components/AddExistingGoalsModal.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        label="Cancel"
+        onTouchTap={() => actions.resetExistingGoalModal()}
+        disabled={parameters.showSpinner}
+      />);
+    const submitButton = (
+      <FlatButton
+        label="Submit"
+        secondary
+        onTouchTap={
+          () => actions.addExistingGoalToPathRequest().then(actions.resetExistingGoalModal())
```
Comment (szymonmichalak) [inline app/components/AddExistingGoalsModal.js]: Maybe let's move this to a separate method like `handleSubmit` or something?

File: app/components/AddExistingGoalsModal.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    return displayedCard;
+  }
+  generateActionButtons() {
+    const { actions, parameters } = this.props;
+    const cancelButton = (
+      <FlatButton
+        label="Cancel"
+        onTouchTap={() => actions.resetExistingGoalModal()}
```
Comment (szymonmichalak) [inline app/components/AddExistingGoalsModal.js]: I would also move this to a separate method like `handleCancel`.

File: app/components/Profile.js [hunk 4]
```diff
@@ -31,11 +44,23 @@ class Profile extends Component {
...
+            paths={paths.list}
+            goals={goals}
+            actions={actions}
+          />
+          <FloatingActionButton
+            style={styles.addExistingButton}
+            onClick={() => actions.showAddExistingGoalsModal(true)}
+          >
+            <ContentPaste />
+          </FloatingActionButton>
```
Comment (szymonmichalak) [inline app/components/Profile.js]: As in #351, I would not suggest to keep more than one floating button on the same page - [Material Design guidelines](https://material.io/guidelines/components/buttons-floating-action-button.html#buttons-floating-action-button-transitions).

Comment (karolsojko) [inline app/components/Profile.js]: As discussed on Slack - we agreed on changing this once we have a design in place.

**Pull request merged**